### PR TITLE
[feat] Support HTTP basic for node client

### PR DESCRIPTION
**Pull request by raymondBourges**
### Motivation

Add HTTP basic for node client

### Modifications

Adaptation of "Feature support oauth2 for node client (#190)" for HTTP basic Authn

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [x] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by raymondBourges** [comment]
Doc is here : https://github.com/apache/pulsar-site/compare/main...raymondBourges:pulsar-site:patch-2

**Comment by raymondBourges** [comment]
Matrix updated : https://github.com/apache/pulsar-site/compare/main...raymondBourges:pulsar-site:patch-3

File: tstest.ts [hunk 1]
```diff
@@ -70,6 +70,11 @@ import Pulsar = require('./index');
     token: 'foobar',
   });
 
+  const authBasic: Pulsar.AuthenticationBasic = new Pulsar.AuthenticationBasic({
+    username: 'basic.username',
+    password: 'basic.password',
+  });
```
Comment (massakam) [inline tstest.ts:76]: There is unnecessary whitespace at the end of the line.
```suggestion
  });
```

**Review by shibd — APPROVED** [review]

**Comment by raymondBourges** [comment]
Thanks @shibd and @massakam,
When do you think 1.12 will be released?

**Comment by shibd** [comment]
Maybe 2 weeks. I'm releasing cpp client. next step is to release node.js client.
